pex/environment.py

```diff
@@ -58,10 +58,19 @@ def update_module_paths(cls, new_code_path):
     TRACER.log('Adding to the head of sys.path: %s' % new_code_path)
     sys.path.insert(0, new_code_path)
     for name, module in sys.modules.items():
-      if hasattr(module, "__path__"):
+      if hasattr(module, '__path__'):
         module_dir = os.path.join(new_code_path, *name.split("."))
         TRACER.log('Adding to the head of %s.__path__: %s' % (module.__name__, module_dir))
-        module.__path__.insert(0, module_dir)
+        try:
+          module.__path__.insert(0, module_dir)
+        except AttributeError:
+          TRACER.log(
```

Comment:
Why is it ok to log and not?:
`# Force subsequent imports to come from the .pex directory rather than the .pex file.`

This begs the question as to whether the enclosing `update_module_paths` function is needed at all.

Reply:
the aim is purely to queue a quick bandaid in the pex->pants->internal release lifecycle to unblock our use of pex here, tbh. I think implicitly allowing breakage of the ~2/100+ affected modules w/ logging beats a hard crash preventing no use until a deeper fix can be made (and subsequently released etc).

this may or may not be reasonable, but just citing my current stance. :)

Reply:
The stance is reasonable, the state the code is left in does not read as reasonable. Can you call the shot in a comment with the real fix issue pointed to?

Reply:
ah, of course.

Reply:
FWIW - I think I sussed what's going on here. Details added to #598. I may be able to get a real fix in here. I'll stare at this code in light of PEP-420 and report an estimate.

Reply:
This looks like it will take a few hours to fix the underlying issue with a test coverging the implicit namespace package case.

Reply:
sgtm - thank you!